### DISCUSSION TITLE
i cant install the code server

### DISCUSSION THREAD

**Discussion by EricAlexandru** [coder/code-server#5917] [Q&A]
When i try to write this command to install it (curl -fsSL https://code-server.dev/install.sh | sh), it says that the term 'sh' is not recognized as the name of a cmdlet, function, script file, or operable program.
What can I do?
![image](https://user-images.githubusercontent.com/78177419/209216119-114a31ea-69e1-436b-b7a6-5b0668ad3b7e.png)

**Comment by jsjoeio** [comment]
Sounds like you might be on Windows? Can you use git bash and try again?

**Comment by lg-nineoneninethreenine** [comment]
This won't work naturally on Windows. Try using WSL or make a Ubuntu Server VM via VirtualBox and try running the command in that.

@jsjoeio Git Bash only gives bash and sh. You also need a package manager to manage the shell, right?

**Reply by jsjoeio** [reply]
Ohhh good point! I didn't realize that.

**Reply by lg-nineoneninethreenine** [reply]
No worries! Git Bash only emulates a Linux shell, it doesn't have a package manager, unlike Cygwin, Mintty and the like(which tbh suck like gutters). WSL and VMs do work, although I would rather suggest a VM over WSL, as WSL is connected to your system, plus Docker doesn't work as expected over there. VMs isolate the files thereby saving the system from possible file corruption and also can run Docker effectively. My only suggestion is to make sure the VM has 1 GB of RAM minimum.
